components/script/dom/url.rs

```diff
@@ -191,6 +202,11 @@ impl URLMethods for URL {
         UrlHelper::SetSearch(&mut self.url.borrow_mut(), value);
```

Comment:
All the setters should update the stored query object if we're going to store it as a field.